### PR TITLE
Add a Gitpod descriptor

### PR DESCRIPTION
**Pull request by pepeiborra**
Develop HLS from your browser in a prebuilt environment.  

Try it with https://gitpod.io/#https://github.com/haskell/haskell-language-server/tree/gitpod

The init script uses ghcup to install GHC and cabal to prebuild all the dependencies of the test suites. 

Using Nix is not an option since the root file system is read-only.

The Free tier allows for 50h monthly hours and the hardware is beefy (>80 cores, 64GB RAM). 

Super handy for testing compatibility with diverse GHC versions.

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    # add a "Review in Gitpod" button to pull requests (defaults to false)
+    addBadge: false
+    # add a label once the prebuild is ready to pull requests (defaults to false)
+    addLabel: prebuilt-in-gitpod
+
+vscode:
+  extensions:
+    - "haskell.haskell"
+    - "justusadam.language-haskell"
+    - "usernamehw.errorlens"
```
Comment (ndmitchell) [inline .gitpod.yml:39]: Do we recommend this for users? Should we?

**Review by ndmitchell — APPROVED** [review]

**Review by jneira — APPROVED** [review]
nice! i tried to add one of this but i was not able to make vscode see cabal nor ghc, maybe cause we did not update the extension in vscodium until recently, many thanks!

**Comment by pepeiborra** [comment]
> nice! i tried to add one of this but i was not able to make vscode see cabal nor ghc, maybe cause we did not update the extension in vscodium until recently, many thanks!

It is a bit tricky, but not impossible hopefully

File: .gitpod.yml (hunk 1 above)
Comment (pepeiborra) [reply .gitpod.yml:49]: Yes, errorlens works great with HLS.

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  - before: |
+      export GHCUP_INSTALL_BASE_PREFIX=/workspace
+      export BOOTSTRAP_HASKELL_NONINTERACTIVE=1
+      export BOOTSTRAP_HASKELL_MINIMAL=1
+      export CABAL_DIR=/workspace/.cabal
+      export STACK_ROOT=/workspace/.stack
+      curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh
+
+      # ensure that there is an HLS binary for the installed GHC version
+      /workspace/.ghcup/bin/ghcup install ghc 8.10.5 --set
```
Comment (Ailrun) [inline .gitpod.yml:12]: Is there a specific reason for using 8.10.5 instead of any later version of GHC?

Comment (pepeiborra) [reply .gitpod.yml:12]: Yes, is the comment not helpful?

Comment (Ailrun) [reply .gitpod.yml:12]: Ah, I misunderstood the comment. I see it now, sorry for bothering you.

**Pull request merged**